app/policies/course_policy.rb

```diff
@@ -1,6 +1,25 @@
 class CoursePolicy
   attr_reader :user, :course
 
+  class Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      if user.admin?
+        scope.all
+      else
+        scope
+          .where(provider_id: user.providers.pluck(:id))
+          .or(scope.where(accrediting_provider_code: user.providers.pluck(:provider_code)))
```

Comment:
I'm not sure that `scope.where(...)` is correct. My understanding is that you need a plain AREL, and not one that already has conditions on it. For example, Course.where(

Here's a quick test:

```ruby
[6] pry(main)> scitts = Course.scitt_programme
[8] pry(main)> scitts.or(scitts.full_time).to_sql
=> "SELECT \"course\".* FROM \"course\" WHERE \"course\".\"program_type\" = 'SC'"
[9] pry(main)> scitts.or(Course.full_time).to_sql
=> "SELECT \"course\".* FROM \"course\" WHERE (\"course\".\"program_type\" = 'SC' OR \"course\".\"study_mode\" = 'F')"
```

In the code above, `scope` is `Course` so the problem doesn't manifest, but I imagine that if we passed in an AREL that's already been scoped, we may get the wrong results.